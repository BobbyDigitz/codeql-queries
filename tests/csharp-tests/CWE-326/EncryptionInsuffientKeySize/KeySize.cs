using System;
using System.IO;
using System.Text;
using System.Runtime.Serialization;
using System.Security.Cryptography;

public class StaticSalt
{
    // public void Test1()
    // {
    //     //
    //     var dsa1 = new DSACryptoServiceProvider();

    //     //
    //     var dsa2 = new DSACryptoServiceProvider(1024);

    // }

    public void TestRSA()
    {
        // BAD: Default is insecure
        var rsa1 = new RSACryptoServiceProvider();
        var rsa2 = new RSACryptoServiceProvider(1024);

        int key_size = 1024;
        var rsa3 = new RSACryptoServiceProvider(key_size);

        // Good
        var rsa_good = new RSACryptoServiceProvider(2048);
    }

    public void TestRSACng()
    {

        // BAD: Default is insecure
        var rsacng = new RSACng();
        var rsacng2 = new RSACng(1024);

        // GOOD: High
        var rsacng_good2 = new RSACng(2048);
    }
}