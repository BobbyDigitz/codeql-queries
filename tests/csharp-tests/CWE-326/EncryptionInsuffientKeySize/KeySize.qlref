CWE-326/EncryptionInsufficientKeySize.ql