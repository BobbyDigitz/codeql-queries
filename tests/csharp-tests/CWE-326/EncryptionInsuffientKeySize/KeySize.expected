| KeySize.cs:22:20:22:49 | object creation of type RSACryptoServiceProvider | Key size 1024 is to small (min: 2048) |
| KeySize.cs:23:20:23:53 | object creation of type RSACryptoServiceProvider | Key size 1024 is to small (min: 2048) |
| KeySize.cs:36:22:36:33 | object creation of type RSACng | Key size 1024 is to small (min: 2048) |
| KeySize.cs:37:23:37:38 | object creation of type RSACng | Key size 1024 is to small (min: 2048) |
